contracts/supplementary/rebalancing-manager/BTCDaiRebalancingManager.sol

```diff
@@ -0,0 +1,475 @@
+        // Create interface to interact with RebalancingSetToken
+        IRebalancingSetToken rebalancingSetInterface = IRebalancingSetToken(_rebalancingSetTokenAddress);
+
+        // Require that enough time has passed from last rebalance
+        uint256 lastRebalanceTimestamp = rebalancingSetInterface.lastRebalanceTimestamp();
+        uint256 rebalanceInterval = rebalancingSetInterface.rebalanceInterval();
+        require(
+            block.timestamp >= lastRebalanceTimestamp.add(rebalanceInterval),
+            "RebalancingTokenManager.proposeNewRebalance: Rebalance interval not elapsed"
+        );
+
+        // Require that Rebalancing Set Token is in Default state, won't allow for re-proposals
+        // because malicious actor could prevent token from ever rebalancing
+        require(
+            rebalancingSetInterface.rebalanceState() == RebalancingHelperLibrary.State.Default,
+            "RebalancingTokenManager.proposeNewRebalance: State must be in Default"
+        );
+
+        // Get price data
+        uint256 btcPrice = queryPriceData();
+
+        // Require that allocation has changed sufficiently enough to justify rebalance
+        uint256 currentSetDollarAmount = checkSufficientAllocationChange(
+            btcPrice,
+            rebalancingSetInterface.currentSet()
+        );
+        
+        // Create new Set Token that collateralizes Rebalancing Set Token
+        (
+            address nextSetAddress,
+            uint256 auctionStartPrice,
+            uint256 auctionPivotPrice
+        ) = createNewAllocationSetToken(
+            btcPrice,
+            currentSetDollarAmount
+        );
+        
+        // Propose new allocation to Rebalancing Set Token
+        rebalancingSetInterface.propose(
+            nextSetAddress,
+            auctionLibrary,
+            auctionTimeToPivot,
+            auctionStartPrice,
+            auctionPivotPrice
+        );
+
+        emit LogManagerProposal(
+            btcPrice
+        );
+    }
+
+    /* ============ Internal ============ */
+    /*
+     * Query price feed for wbtc, return as uint256
+     *
+     * @return          BTC price
+     */
+    function queryPriceData()
+        private
+        view
+        returns (uint256)
+    {
+        // Get price from oracles
+        bytes32 btcPriceBytes = IMedian(btcPriceFeed).read();
+
+        // Cast bytes32 prices to uint256
+        uint256 btcPrice = uint256(btcPriceBytes);
+
+        return (btcPrice);
+    }
+
+    /*
+     * Check there has been a sufficient change in allocation as defined by maximumUpperThreshold
+     * and minimumLowerThreshold and return USD value of currentSet.
+     *
+     * @param  _btcPrice              The 18 decimal dollar value of one full BTC
+     * @param  _currentSetAddress     The address of the Rebalancing Set Token's currentSet
+     * @return                        The currentSet's USD value (in cents)
+     */
+    function checkSufficientAllocationChange(
```

Comment:
I'm imagining there's a way to dedupe this code across managers through a library